File: packages/api/server/lib/shared/document_draft.go
package shared

type DocumentDraft struct {
	Entity

	DocumentId  string           `json:"documentId"`
	Name        string           `json:"name"`
	Content     *DocumentContent `json:"content,omitempty"`
	CreatorId   string           `json:"creatorId"`
	PublishedAt *int64           `json:"publishedAt"`
	RetractedAt *int64           `json:"retractedAt"`
}

File: packages/frontend-shared/src/store/model/document/document-draft.ts
import {Entity} from "../entity";
import {DocumentContent} from "./document-content";
import {Exclude, Expose, Type} from "class-transformer";

@Exclude()
export class DocumentDraft extends Entity {

    @Expose()
    public name: string;

    @Expose()
    public documentId: string;

    @Expose()
    public publishedAt: number | null;

    @Expose()
    public retractedAt: number | null;

    @Expose()
    @Type(() => DocumentContent)
    public content: DocumentContent | null;

}
